tracing-appender/src/inner.rs

```diff
@@ -75,6 +77,87 @@ impl InnerAppender {
     }
 }
 
+#[derive(Debug)]
+pub(crate) struct InnerRotatingAppender {
+    log_directory: String,
+    log_filename_prefix: String,
+    writer: BufWriter<File>,
+    last_backup: usize,
+    current_size: usize,
+    rotation: Rotation,
+}
+
+impl io::Write for InnerRotatingAppender {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        let buf_len = buf.len();
+        self.refresh_writer(&buf_len);
+        self.writer.write_all(buf).map(|_| buf_len)
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        self.writer.flush()
+    }
+}
+
+impl InnerRotatingAppender {
+    pub(crate) fn new(
+        rotation: Rotation,
+        log_directory: &Path,
+        log_filename_prefix: &Path,
+    ) -> io::Result<Self> {
+        let log_directory = log_directory.to_str().unwrap();
+        let log_filename_prefix = log_filename_prefix.to_str().unwrap();
+        let current_size = get_file_size(log_directory, log_filename_prefix)?;
+        let last_backup = Self::find_last_backup(&rotation, log_directory, log_filename_prefix);
+        Ok(Self {
+            writer: create_writer(log_directory, log_filename_prefix)?,
+            log_directory: log_directory.to_string(),
+            log_filename_prefix: log_filename_prefix.to_string(),
+            last_backup,
+            current_size,
+            rotation,
+        })
+    }
+    fn refresh_writer(&mut self, size: &usize) {
```

Comment:
nit: A new line between methods could be added in a few places.

Reply:
will fix this.

Reply:
done